src/masternode/activemasternode.cpp

```diff
@@ -68,7 +68,7 @@ void CActiveMasternodeManager::Init(const CBlockIndex* pindex)
     if (!deterministicMNManager->IsDIP3Enforced(pindex->nHeight)) return;
 
     // Check that our local network configuration is correct
-    if (!fListen) {
+    if (!fListen && Params().NetworkIDString() != CBaseChainParams::REGTEST) {
```

Comment:
Really hate seeing `Params().NetworkIDString() != CBaseChainParams::REGTEST` kind of stuff in code tbh. How about
```suggestion
    if (!fListen && fRequireRoutableExternalIP) {
```
?

Reply:
Agree. I added a commit that replaces the REGTEST checks in all places I found (not only the ones related to this PR).